### ISSUE TITLE
Environment variables read by build logic depends on JDK used and daemon setting

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle#3468]
When using JDK9, `System.getenv` does not reflect the environment the build was started from unless the daemon is disabled and a new JVM is used on each build.  On JDK7&8, we copy the environment variables from the client JVM to the daemon process.

We stopped doing that for JDK9:
https://github.com/gradle/gradle/blob/master/subprojects/native/src/main/java/org/gradle/internal/nativeintegration/processenvironment/AbstractProcessEnvironment.java#L81-L87

### Expected Behavior

From build logic, I have a way to get values from the environment Gradle was run from.

### Current Behavior

`System.getenv` can be used for JDK7 and 8.  For JDK9, I have to disable the daemon completely or avoid the use of environment variables.

### Context

This is impacting gradle-native: https://github.com/gradle/gradle-native/issues/280

This will probably impact more things going forward.

### Steps to Reproduce (for bugs)

With JDK9, given `build.gradle`:
```
task printVar() {
  doLast { 
    println System.env.VAR
  }
}
```

```
export VAR=foo
gradle printVar // prints foo
export VAR=bar
gradle printVar // prints foo
```

### Your Environment

Any OS, JDK9+, any Gradle that supports JDK9.

**Comment by oehme** [comment]
Java 9 does not allow us to overwrite the environment anymore. 

We could ship them over via a dedicated channel and offer something like `StartParameter.getEnvironment`. Or we could expand our new build option infrastructure to be extensible and also be configurable from the environment. In both cases we need to ship the environment to the daemon in some form. The difference is just how we surface that in the API.

**Comment by big-guy** [comment]
Yep, I tried to avoid prescribing a solution (e.g., `System.getenv` should _just work_).

**Comment by GFriedrich** [comment]
Hi,
@oehme is it still planned to add a `StartParameter.getEnvironment`?
Because I've the problem that my CI server uses some environment variables to hand over some information of the build itself. Those variable are read via `System.env`, but as I use the Gradle daemon the second and subsequent build are using the old and wrong variables.
Of course I could add all of those variables to the command line but that would make the command awefully long.
So I would really appreciate if you could add such a new method. What do you think @oehme ?
Thanks in advance.

**Comment by oehme** [comment]
Fixed by #6170
